monai/transforms/utils.py

```diff
@@ -157,18 +157,16 @@ def rescale_array(
     """
     Rescale the values of numpy array `arr` to be from `minv` to `maxv`.
     """
-    dtype_ = dtype or arr.dtype
+    if dtype is not None:
+        arr, *_ = convert_data_type(arr, dtype=dtype)
     mina = arr.min()
     maxa = arr.max()
 
     if mina == maxa:
         return arr * minv
 
     norm = (arr - mina) / (maxa - mina)  # normalize the array first
-    arr = (norm * (maxv - minv)) + minv  # rescale by minv and maxv, which is the normalized array by default
-
-    ret, *_ = convert_data_type(arr, dtype=dtype_)
-    return ret
+    return (norm * (maxv - minv)) + minv  # rescale by minv and maxv, which is the normalized array by default
```

Comment:
Hi @ericspod @wyli ,

I think this change is not enough, it will cause below behavior:
```py
>>> from monai.transforms import rescale_array
>>> 
>>> 
>>> from monai.transforms import rescale_array
>>> import numpy as np
>>> 
>>> a = np.array([1, 2, 3])
>>> 
>>> b = rescale_array(a, minv=80, maxv=100, dtype=None)
>>> 
>>> print(a.dtype, b.dtype)
int64 float64
```

Let me try to enhance it ASAP.

Thanks.

Reply:
I think the current one is fine for this utility, when dtype is none, it fallbacks to  the numpy-like behaviour.

Reply:
Hi @wyli @ericspod ,

Current one doesn't work as expected for some cases not only `dtype=None`, for example:
```py
>>> from monai.transforms import rescale_array
>>> import numpy as np
>>> 
>>> a = np.array([1, 2, 3])
>>> b = rescale_array(a, minv=80, maxv=100, dtype=np.uint8)
>>> 
>>> print(a.dtype, b.dtype)
int64 float64
```

I submitted an enhance: https://github.com/Project-MONAI/MONAI/pull/3246, could you please help take a look?

Thanks in advance.